torchgeo/datasets/cropharvest.py

```diff
@@ -0,0 +1,317 @@
+from .utils import download_and_extract_archive, download_url
+
+
+class CropHarvest(NonGeoDataset):
+    """CropHarvest dataset.
+
+    The CropHarvest <https://github.com/nasaharvest/cropharvest> datataset is a
+    classification dataset.
+
+    Dataset features:
+
+    * _ pixel single timeseries with croptype labels
+    * 18 bands per image over 12 months
+
+    Dataset format:
+
+    * images are 12x18 ndarrays with 18 bands over 12 months
+
+    Dataset properties:
+
+    1. is_crop - whether or not a single pixel contains cropland
+    2. classification_label - optional field identifying a specific croptype
+    3. dataset - source datset for the imagery
+    4. lat
+    5. lon
+
+
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://openreview.net/forum?id=JtjzUXPEaCu
+
+    This dataset requires the following additional library to be installed:
+
+       * `h5py <https://pypi.org/project/h5py/>`_ to load the dataset
+    """
+
+    # *https://github.com/nasaharvest/cropharvest/blob/main/cropharvest/bands.py
+    all_bands = [
+        "VV",
+        "VH",
+        "B2",
+        "B3",
+        "B4",
+        "B5",
+        "B6",
+        "B7",
+        "B8",
+        "B8A",
+        "B9",
+        "B11",
+        "B12",
+        "temperature_2m",
+        "total_precipitation",
+        "elevation",
+        "slope",
+        "NDVI",
+    ]
+    rgb_bands = ["B11", "B8", "B2"]
+
+    features_url = "https://zenodo.org/records/7257688/files/features.tar.gz?download=1"
+    labels_url = "https://zenodo.org/records/7257688/files/labels.geojson?download=1"
+    file_dict = {
+        "features": {
+            "url": features_url,
+            "filename": "features.tar.gz",
+            "extracted_filename": os.path.join("features", "arrays"),
+            "md5": "cad4df655c75caac805a80435e46ee3e",
+        },
+        "labels": {
+            "url": labels_url,
+            "filename": "labels.geojson",
+            "extracted_filename": "labels.geojson",
+            "md5": "bf7bae6812fc7213481aff6a2e34517d",
+        },
+    }
+
+    directory = "CropHarvest"
+
+    def __init__(
+        self,
+        root: str = "data",
+        transforms: Optional[Callable[[dict[str, object]], dict[str, object]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new CropHarvest dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train" or "val"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
```

Comment:
See #1714, we added a new error type specifically for this case that you should use